src/Modal.jsx

```diff
@@ -31,6 +36,28 @@ class Modal extends React.Component {
 		}
 	}
 
+	/**
+	 * @param {String} scrollPosition window scroll position
+	 * @returns {String} CSS value for setting modal margin-top
+	 */
+	getModalPosition(scrollPosition) {
+		const viewportHeight = Math.max(document.documentElement.clientHeight, window.innerHeight || 0);
+
+		if (scrollPosition < viewportHeight) {
+			return DEFAULT_MARGIN_TOP;
```

Comment:
In CSS, the default `margin-top` for a snap modal is `10vh`. The CSS also removes this margin if the viewport is below the `medium` breakpoint. 

If we set `margin-top` programmatically from the component, the component also needs to be aware of when we're below the `medium` breakpoint so that we can avoid forcing a margin on the modal when the viewport is small and the modal should be full screen.